src/main/java/net/glowstone/net/handler/play/player/UseItemHandler.java

```diff
@@ -90,6 +92,10 @@ static void handleRightClickAir(GlowPlayer player, ItemStack holding, EquipmentS
             if (type != null) {
                 if (type.getContext().isAirApplicable()) {
                     type.rightClickAir(player, holding);
+                } else if (type instanceof ItemTimedUsage) {
+                    ((ItemTimedUsage) type).startUse(player, holding);
```

Comment:
This call is already in the `rightClickAir` handler of `ItemTimedUsage`. You should make ItemProjectile similar to ItemTimedUsage (returning `Context.ANY` for `getContext()` and calling the `use` method in both `rightClickAir` and `rightClickBlock`). I haven't tested it, but I think currently you are not able to shoot a projectile when aiming at a block.

Reply:
Done in 985f213b34b6752c1058d5d2f37d824a9eacaed3